opm/grid/common/WellConnections.cpp

```diff
@@ -82,13 +82,23 @@ WellConnections::WellConnections(const std::vector<OpmWellType>& wells,
                                  const std::unordered_map<std::string, std::set<int>>& possibleFutureConnections,
                                  const Dune::CpGrid& cpGrid)
 {
-    const auto& cpgdim = cpGrid.logicalCartesianSize();
+    // - If the grid with dimensions {NX, NY, NZ} has been globally refined 'count' times,
+    //   then cpgrid.logicalCartesianSize() returns {2^count * NX, 2^count * NY, 2^count * NZ}.
+    //   By default, global refinement subdivides each cell into 2 in each direction (x, y, and z).
+    //
+    // - If the grid has undergone local (not global) refinement, cpgrid.logicalCartesianSize()
+    //   returns the logical Cartesian size of the original (level-zero) grid.
+    const auto& cpgdim = cpGrid.maxLevel()? cpGrid.currentData().front()->logicalCartesianSize() : cpGrid.logicalCartesianSize();
```

Comment:
I think both branches should do the same, right? Maybe skip the decision and just always use the first?

Reply:
There is an inconsistency in how logical Cartesian size behaves in the following cases:
1. global refinement (well-defined for the leaf grid view)
2. local refinement via CARFIN (startIJK and endIJK) -> each refined level grid gets a well-defined logical Cartesian size. However, the leaf grid view gets a copy of level zero grid's logical Cartesian size.
3. local refinement via adapt() -> refined level grids and leaf grid get a copy of level zero grid's logical Cartesian size. 

I added a global refine case in the test, and deleted the comment regarding logical Cartesian size different behavior, since I was thinking to add a separated test (in a separated PR) describing this inconsistency. 

In any case, we can just take level zero logical Cartesian size here, omitting the decision.